src/Xamarin.Android.Build.Tasks/Tasks/GenerateMergedAndroidManifest.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Java.Interop.Tools.TypeNameMappings;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Framework;
+using Mono.Cecil;
+using Xamarin.Android.Tools;
+
+namespace Xamarin.Android.Tasks;
+
+public class GenerateMergedAndroidManifest : AndroidTask
```

Comment:
Some of this can be trimmed down since we use the android manifestmerger these days. 

``AndroidManifestMerger` == `legacy`` is what this is keyed off. 
But I guess this will be a staged approach. 
Maybe we can rename this task a bit? Because unless we use legacy we use another tool to generate the final manifest. Perhaps `g` ?

Reply:
I don't see anything that prevents this logic from running if `$(AndroidManifestMerger) != 'legacy'`. If that is the intention, then that is definitely something we should look into in the future.

I can rename the task `GenerateLegacyMergedAndroidManifest` to help make the purpose clearer.

Reply:
Well it is partially used in the new system. The main difference is the `_MergedManifestDocuments` ItemGroup is not populated in the new system. So that will be empty.

Reply:
See https://github.com/dotnet/android/blob/b2c73ff70858e83c0303ef0de614f51b8992c6d1/src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets#L1512

Reply:
Hmm, then calling it `GenerateLegacyMergedAndroidManifest` is probably not a good name. Do you have a suggestion?

Reply:
maybe `GenerateIntermediateAndroidManifest` ?

Reply:
or `GenerateMainAndroidManifest` since its the "root" manifest when we finally do the merging later.